lib/github/hooks.nit

```diff
@@ -0,0 +1,172 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Github hook event listening with `nitcorn`.
+#
+# Usage:
+#
+# ~~~
+# import github::hooks
+#
+# # A simple hook listener that print received events in stdout.
+# class LogHookListener
+#	super HookListener
+#
+#	redef fun api: GithubAPI do return new GithubAPI(get_github_oauth)
+#
+#	redef fun apply_event(event) do
```

Comment:
peut-etre dans la doc proposer de passer par un double dispatch plutot que par un tas de if, non?

~~~
class LogHookListener
   super HookListener
   redef fun apply_event(event) do event.log_event(self)
end
redef class GithubEvent # la super-classe
   fun log_event(l: LogHookListener) is do end # by defaut do nothing
end
redef class CommitCommentEvent
   redef fun log_event(l) do des trucs...
end
~~~